### DISCUSSION TITLE
New Field Type : Duration

### DISCUSSION THREAD

**Discussion by poirazis** [pods-framework/pods#6800] [Ideas]
## Problem

There are times when you need to save the Duration of say an event. The Time field cannot cater to this need, as the it's purpose is to pinpoint to a specific time in the 24 day. The Duration Field would be able to hold values of any Duration of time. For example assume we have an event that lasted 3days 1hr 35min. 

Lets say we have a field EventStart : DateTime and an EventEnd : DateTime, the EventDuration field would be able to hold the value of EventEnd - EventStart. 

Another example would be when we have a lets say Event like Burning Man. We would be able to set the Duration of the Event.

## Proposed Solution

The proposed solution would be to add a Duration Field type which will keep its value in seconds elapsed and when displayed the value would be presented formatted in different ways. 

This could be core functionality, or part of an Pods Extended Field Types Component as a collection of additional Field Types.

**Comment by sc0ttkclark** [comment]
I like the idea here, I'm sure this is going to be complicated to figure out how to best do this from an input standpoint.

Do we have text on the page that represents the value, with a link to edit the value and a text field that you can enter freeform text into that tries to get translated (and has validation that warns you if it's not valid) -- or do we have some sort of series of number inputs for years, months, days, hours, minutes, seconds?

**Comment by poirazis** [comment]
hey,
my initial thought would be o have a series of number inputs. The idea was to have 3 number input field ( H M S ) and a + sign on the left. When the + was clicked a new input box (of higher order) would appear. So at the first + click we wouls add Day, then + again would add Week etc. 

(sorry for the delay - I am in GMT +2)

**Comment by sc0ttkclark** [comment]
Have you seen this done anywhere else that you liked?

**Comment by poirazis** [comment]
This is done in MatLAB like this https://www.mathworks.com/help/matlab/ref/duration.html#responsive_offcanvas
I haven't seen the perfect implementation yet!
